### ISSUE TITLE
nightly doesn't reconize certain commands

### ISSUE THREAD

**Issue by imrgrim** [ikemen-engine/Ikemen-GO#2159]
### Describe the bug

Some characters can't perform half circle forward i've tested those character in version 0.96 and 0.99 and they work fine but not in nightly the characters are vice by jmorph, k dash byChazzanova & Dampir and ryu by raginrowen in kdash's case he can't also perform his chain drive 

### To Reproduce

1. select the character
2. perform a half circle forward + kick or punch motion
3. nothing happends

### Engine Version (or source code date)

Mar 9, 2023

### Operating system

Windows

### Extra context or search terms

I'm using keyboard haven't tested this with controller nor stick

**Comment by potsmugen** [comment]
Can't reproduce. Did you modify the chars to work better in old Ikemen? Can you do half circle back?

**Comment by imrgrim** [comment]
> Can't reproduce. Did you modify the chars to work better in old Ikemen? Can you do half circle back?

No, i haven't modify them and nope i can't do half circle back but it's weird because only those characters aren't working i've tested most of my characters and It seems onyl those characters refuse to work

**Comment by imrgrim** [comment]
> Can't reproduce. Did you modify the chars to work better in old Ikemen? Can you do half circle back?

And as I said they were working perfectly fine in previous versions

**Comment by imrgrim** [comment]
I fix some of the chars by replacing the commands for example half circle forward was ~B, DB, D, DF, F now ~$DB, $F, I've also tested the same characters in nightly using an arcade stick and they work without changing anything so it must be an error related to keyboard compatibility another command that doesn't work correctly is quarter circle forward quarter circle back (or viceversa) this one kinda works but isn't as smooth as it used to be you need to be extremely precise, before i would just do quarter circle forward, back without inputing the full quarter circle back

**Comment by potsmugen** [comment]
Can you do all of Kung Fu Man's moves?

**Comment by imrgrim** [comment]
> Can you do all of Kung Fu Man's moves?

Yes except for kungfu upper but i think is just an error of the command list

**Comment by imrgrim** [comment]
I recorded my inputs on 3 different versions, both 0.99 and nightly are clean and the character is the same in all 3 versions (Ralf by Infinite) the command are more responsive in 0.96 and 0.99 but are extremely difficult to perform in nightly and half circle forward doesn't even work

https://youtu.be/mg9MMvC_SRA  0.96

https://youtu.be/NCnsRunEQJc  0.99

https://youtu.be/s3JqtxStNJU  Nightly

**Comment by potsmugen** [comment]
Can you upload or link the exact version of Ralf you used? Are you using a secondary keyboard?

**Comment by imrgrim** [comment]
https://www.mediafire.com/folder/56ssxkhlyoxuy/Ralf 

https://www.mediafire.com/folder/q5t2fqoikxbv0/Kyo

the first link is for Ralf the second is for kyo i send kyo because he's the one of the few chars that works in all versions with the same commands

**Comment by potsmugen** [comment]
You mind reuploading as a single zip archive? Currently we need to download dozens of files one by one.

**Comment by imrgrim** [comment]
https://www.mediafire.com/file/nquw60sy6g3siv8/Chars.zip/file

**Comment by potsmugen** [comment]
It's hard to be objective when it comes to all sorts of different hardware, but here's a recording where Ikemen and Mugen are both controlled simultaneously by the same pad.

https://github.com/user-attachments/assets/30009db4-5ee2-4493-a18d-fea0463417c8

They do fall out of sync every now and then but I drop the inputs in Ikemen as often as in Mugen.

Do you have a gaming keyboard or something? I think there have been some changes to hardware compatibility lately.

Also try configuring the directions to the arrows. It could be a ghosting issue.

**Comment by imrgrim** [comment]
Yes i'm using a gaming keyboard more specifically a Logitech G915 TKL keyboard, i've tried changing the directions to the arrows but same problem

**Comment by potsmugen** [comment]
I took a closer look at how you do the inputs on keyboard (thanks for recording them). Try opening your config.json file and changing `InputSOCDResolution` to 4.

**Comment by imrgrim** [comment]
> I took a closer look at how you do the inputs on keyboard (thanks for recording them). Try opening your config.json file and changing `InputSOCDResolution` to 4.

YEEEEES IT SOLVED MY PROBLEM!!! THANK YOU SO MUUUCH

**Comment by potsmugen** [comment]
4 ignores B and F if you press them simultaneously. Normally that makes inputs more strict, but I noticed you have the habit of pressing B, D and F together when doing half circles so it compensates for that.

**Comment by K4thos** [comment]
4 is the new default, so I'm closing this ticket

**Comment by potsmugen** [comment]
I have mixed feelings about changing the default. One one hand, the default has been 2 for roughly a year without prior issues. On the other Street Fighter has moved from "2" to "4" between 5 and 6, so we're kind of keeping with the times.

I guess at the end of the day what matters is that it can be configured. Hopefully some day we'll figure out the netplay side of it.
